app/controllers/expenses_controller.rb

```diff
@@ -0,0 +1,74 @@
+class ExpensesController < ApplicationController
+  before_action :authenticate_user!, except: [:splash]
+  before_action :set_expense, only: %i[show edit update destroy]
+  before_action :set_category
+
+  # GET /expenses or /expenses.json
+  def index
+    @expenses = Expense.all
+  end
+
+  # GET /expenses/1 or /expenses/1.json
+  def show; end
+
+  # GET /expenses/new
+  def new
+    @expense = Expense.new
+  end
+
+  # GET /expenses/1/edit
+  def edit; end
+
+  # POST /expenses or /expenses.json
+  def create
+    @expense = Expense.new(expense_params)
+    respond_to do |format|
+      if @expense.save
+        format.html { redirect_to expense_url(@expense), notice: 'Expense was successfully created.' }
+        format.json { render :show, status: :created, location: @expense }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @expense.errors, status: :unprocessable_entity }
+      end
+    end
+  end
```

Comment:
- [ ] Great job implementing the association for category belongs to use. But your web app still does not work as expected. The user is still not able to create a transaction. The error is shown in the screenshot below states that there is no user assigned for the transaction. I recommend adding the current user when you create a transaction so that the transaction belongs to the current user. There are many ways to implement that. I recommend looking at the example below to get an idea about this:
```rb
@expense = current_user.expenses.new(expense_params)
```
It's ok if use another solution.  You just need to ensure that the transaction is created successfully and belongs to the current user.
- The error:
<img width="202" alt="image" src="https://user-images.githubusercontent.com/77190203/209538711-663cea0e-0fe5-4050-967d-f90bddad44e7.png">
 
Note: I recommend doing the same thing for creating a category. A category should belong to the current user when it's created.